dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/export/event/EventChangeLogServiceTest.java

```diff
@@ -463,9 +459,225 @@ void shouldReturnEventPropertiesChangeLogWhenExistingGeometryPointPropertyDelete
             assertPropertyCreate("geometry", "(-11.419700, 8.103900)", geometryChangeLogs.get(1)));
   }
 
-  private void updateDataValue(String event, String dataElementUid, String newValue)
-      throws IOException {
-    TrackerObjects trackerObjects = fromJson("tracker/event_and_enrollment.json");
+  private static Stream<Arguments> provideOrderParamsDesc() {
+    return Stream.of(
+        Arguments.of(
```

Comment:
All the cases have

```java
            Arrays.asList("25", "20", "15"),
            Arrays.asList("20", "15", null)),
```

so they should not be a parameter as they are always the same. Right now I had to figure that out which is clear if you put them as is into the test.

I hope this would also make it easy to compare the expectations with the inverse (reverse order) test of `shouldSortChangeLogsWhenOrderingByCreatedAtAsc`.

Reply:
I agree with the first statement, after the last refactor they all ended up being the same, so it makes no sense to have those arguments anymore.
However, the reason why I couldn't combine it with `shouldSortChangeLogsWhenOrderingByCreatedAtAsc`, is because I want to test the `changeLogType`. Unless I do something very ugly in the test, I couldn't come up with a solution to test the four cases together.